Sources/firebase/include/FirebaseFirestore.hh

```diff
@@ -44,6 +44,37 @@ document_get(const ::firebase::firestore::DocumentReference document, ::firebase
   return document.Get(source);
 }
 
+inline ::firebase::firestore::DocumentReference
+snapshot_reference(const ::firebase::firestore::DocumentSnapshot snapshot) {
+  return snapshot.reference();
+}
+
+struct MapFieldValue_Workaround {
```

Comment:
Could you guard the workaround with a `#if SR?????` substituting the issue number so we could clean it up?

Reply:
Sure, what's `SR`?